plugins/chat/app/services/chat/auto_leave_channels.rb

```diff
@@ -0,0 +1,87 @@
+# frozen_string_literal: true
+
+module Chat
+  class AutoLeaveChannels
```

Comment:
This service is part 2/2 of the main change introduced in this PR.

It efficiently handles automatically leaving users.